plugins/action/action_impl.cpp

```diff
@@ -133,17 +133,20 @@ ActionResult ActionImpl::return_to_launch() const
         _parent->set_flight_mode(SystemImpl::FlightMode::RETURN_TO_LAUNCH));
 }
 
-ActionResult ActionImpl::goto_location(float latitude, float longitude, float altitude)
+ActionResult ActionImpl::goto_location(double latitude, double longitude, double altitude)
 {
+    float lat1e7 = (float) (latitude * 1E7);
```

Comment:
you don't need this. You can use a floating point literal:
```c++
ActionResult ActionImpl::goto_location(float latitude, float longitude, float altitude)
{
        ....
        command.params.param5 = latitude * 1E7F;
        command.params.param6 =  longitude * 1E7F;
        ....
}
```

Reply:
if I keep `latitude` and `longitude` as `float`, I get warnings when I enter big values like `48.1234567` for loosing precision.